### PR TITLE
Converting origin_tech to JSON from lists.

### PR DESCRIPTION
**Pull request by MistakeNot4892**
- [x] Convert with regex.
- [x] Test/debug.
- [x] Add unit test.

### CODE REVIEW

File: code/modules/projectiles/guns/magnetic/magnetic_railgun.dm [hunk 1]
```diff
@@ -5,7 +5,7 @@
 	icon_state = "railgun"
 	removable_components = TRUE // Can swap out the capacitor for more shots, or cell for longer usage before recharge
 	load_type = /obj/item/rcd_ammo
-	origin_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 4, TECH_MAGNET = 4)
+	origin_tech = "{'" + TECH_COMBAT + "':5,'" + TECH_MATERIAL + "':4,'" + TECH_MAGNET + "':$6}"
```
Comment (SodiumShine) [inline code/modules/projectiles/guns/magnetic/magnetic_railgun.dm]: $6 worth of railgun

File: code/modules/projectiles/guns/magnetic/magnetic_railgun.dm [hunk 2]
```diff
@@ -84,7 +84,7 @@
 	load_type = /obj/item/stack/material/rods // The Confederation learned that chunks of metal work just as well as fancy matter cartridges - actually they dont
 	projectile_type = /obj/item/projectile/bullet/magnetic
 	load_sheet_max = 6 // Fewer shots per "magazine", but more abundant than matter cartridges.
-	origin_tech = list(TECH_COMBAT = 6, TECH_MATERIAL = 3, TECH_MAGNET = 5)
+	origin_tech = "{'" + TECH_COMBAT + "':6,'" + TECH_MATERIAL + "':3,'" + TECH_MAGNET + "':$6}"
```
Comment (SodiumShine) [inline code/modules/projectiles/guns/magnetic/magnetic_railgun.dm]: $6

File: code/modules/projectiles/guns/projectile/automatic.dm [hunk 3]
```diff
@@ -33,7 +33,7 @@
 	safety_icon = "safety"
 	item_state = "mpistolen"
 	caliber = CALIBER_PISTOL
-	origin_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 2, TECH_ESOTERIC = 3)
+	origin_tech = "{'" + TECH_COMBAT + "':5,'" + TECH_MATERIAL + "':2,'" + TECH_ESOTERIC + "':$6}"
```
Comment (SodiumShine) [inline code/modules/projectiles/guns/projectile/automatic.dm]: $

Comment (MistakeNot4892) [reply code/modules/projectiles/guns/projectile/automatic.dm]: Thanks regex

File: code/modules/projectiles/guns/projectile/automatic.dm [hunk 4]
```diff
@@ -70,7 +70,7 @@
 	w_class = ITEM_SIZE_LARGE
 	force = 10
 	caliber = CALIBER_PISTOL
-	origin_tech = list(TECH_COMBAT = 5, TECH_MATERIAL = 2, TECH_ESOTERIC = 8)
+	origin_tech = "{'" + TECH_COMBAT + "':5,'" + TECH_MATERIAL + "':2,'" + TECH_ESOTERIC + "':$6}"
```
Comment (SodiumShine) [inline code/modules/projectiles/guns/projectile/automatic.dm]: $

**Pull request merged**